src/test/java/testcases/TestBase.java

```diff
@@ -2,11 +2,29 @@
 
 import drivers.DriverUtils;
 import org.testng.annotations.AfterMethod;
+import org.testng.annotations.DataProvider;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+
+import static utils.YmlParser.getTestData;
 
 
 public class TestBase {
 
-    @AfterMethod
+    @DataProvider(name = "dataByMethod")
+    public Object[][] getTestDataFromYaml(Method method) throws Exception {
+        String testCaseName = method.getName();
+        List<Map<String, String>> dataList = getTestData(testCaseName);

```

Comment:
In case you **really really** need parse test data from file, I suggest you create a class then use snakeyaml to deserialize it